mam/v2/api/api.c

```diff
@@ -166,12 +168,60 @@ retcode_t mam_api_bundle_write_packet(mam_api_t *const api,
   return RC_OK;
 }
 
-retcode_t mam_api_bundle_read(mam_api_t *const api,
-                              mam_channel_t const *const cha,
-                              bundle_transactions_t const *const bundle,
-                              flex_trit_t *const payload) {
-  if (api == NULL || cha == NULL || bundle == NULL || payload == NULL) {
-    return RC_NULL_PARAM;
+bool mam_api_bundle_contains_header(bundle_transactions_t const *const bundle) {
+  iota_transaction_t *tx = (iota_transaction_t *)utarray_front(bundle);
+  return flex_trits_are_null(transaction_tag(tx), NUM_TRITS_TAG);
+}
+
+retcode_t mam_api_bundle_read_msg(mam_recv_msg_context_t *const ctx,
+                                  mam_channel_t const *const cha,
+                                  bundle_transactions_t const *const bundle,
+                                  flex_trit_t **const packet_payload,
+                                  trits_t session_key, trits_t msg_id) {
```

Comment:
Why is session_key a parameter ?

Reply:
because when you want to read the rest of the packets that belong to that msg, you need it for decryption

Reply:
You don't actually need it. You absorb it into your spongos state and you just need to keep your spongos.

Reply:
Ok then :)

Reply:
everything in MAM is soo cryptic

Reply:
:trollface: